src/app/routes/api/v3/forms/__tests__/public-forms.auth.routes.spec.ts

```diff
@@ -279,4 +288,182 @@ describe('public-form.auth.routes', () => {
+        errorCode: '138',
+      })
+
+      // Act
+      const response = await request.get(`/forms/${form._id}/auth/validate`)
+
+      // Assert
+      expect(response.status).toEqual(200)
+      expect(response.body).toEqual(expectedResponse)
+    })
+
+    it('should return 400 when the form has an invalid AuthType', async () => {
+      // Arrange
+      const { form } = await dbHandler.insertEmailForm({
+        formOptions: {
+          authType: AuthType.NIL,
+          esrvcId: new ObjectId().toHexString(),
+          status: Status.Public,
+        },
+      })
+      const expectedResponse = jsonParseStringify({
+        message:
+          'Please ensure that the form has authentication enabled. Please refresh and try again.',
+      })
+
+      // Act
+      const response = await request.get(`/forms/${form._id}/auth/validate`)
+
+      // Assert
+      expect(response.status).toEqual(400)
+      expect(response.body).toEqual(expectedResponse)
+    })
+
+    it('should return 400 when the form does not have an eSrvcId', async () => {
+      // Arrange
+      const { form } = await dbHandler.insertEmailForm({
+        formOptions: {
+          authType: AuthType.SP,
+          status: Status.Public,
+        },
+      })
+      const expectedResponse = jsonParseStringify({
+        message:
+          'This form does not have a valid eServiceId. Please refresh and try again.',
+      })
+
+      // Act
+      const response = await request.get(`/forms/${form._id}/auth/validate`)
+
+      // Assert
+      expect(response.status).toEqual(400)
+      expect(response.body).toEqual(expectedResponse)
+    })
+
+    it('should return 404 when the form does not exist', async () => {
+      // Arrange
+      const expectedResponse = jsonParseStringify({
+        message:
+          'Could not find the form requested. Please refresh and try again.',
+      })
+
+      // Act
+      const response = await request.get(
+        `/forms/${new ObjectId().toHexString()}/auth/validate`,
+      )
+
+      // Assert
+      expect(response.status).toEqual(404)
+      expect(response.body).toEqual(expectedResponse)
+    })
+
+    it('should return 502 when the fetched login page has no title', async () => {
+      // Arrange
+      const { form } = await dbHandler.insertEmailForm({
+        formOptions: {
+          authType: AuthType.SP,
+          esrvcId: new ObjectId().toHexString(),
+          status: Status.Public,
+        },
+      })
+      MockAxios.get.mockResolvedValueOnce({
+        data: '',
+      })
+      const expectedResponse = jsonParseStringify({
+        message: 'Error while contacting SingPass. Please try again.',
+      })
+
+      // Act
+      const response = await request.get(`/forms/${form._id}/auth/validate`)
+
+      // Assert
+      expect(response.status).toEqual(502)
+      expect(response.body).toEqual(expectedResponse)
+    })
+
+    it('should return 500 when a database error occurs', async () => {
```

Comment:
appreciate the effort to mock the services, but bear in mind that this isn't strictly necessary in integration tests